### DISCUSSION TITLE
Is Combo box supported ?

### DISCUSSION THREAD

**Discussion by eternity74** [wezterm/wezterm#4743] [Q&A] [answered]
### What Operating System(s) are you running on?

Windows

### WezTerm version

20231203-124028-e3cd2e93

### Ask your question!

Hi, 
I really like this terminal for daily use especially customizing for my usage.
I would like to setup menu like command palette for creating ad-hoc ssh tab.
I use multiple ssh connection to multiple embedded devices and their ip address is not static.
So I want to setup list of ip adress or hostname and select or input from combobox and then make it ad-hoc ssh connection on new tab.
I'm trying to find a way to achive this but I haven't found any solution yet.
So I have two questions.
1. Is combobox supported ? if not, is there any possible way to achive same behavior ?
2. is there any way to use ad-hoc ssh connection with arbitrary ip address ?

Thanks for providing such a good proejct.

**Comment by nekowinston** [comment]
WezTerm can parse your SSH config `~/.ssh/config`, which you can further modify with [`enumerate_ssh_hosts`](https://wezfurlong.org/wezterm/config/lua/wezterm/enumerate_ssh_hosts.html).
To quickly connect to ssh targets, I like to use the [Command Palette](https://wezfurlong.org/wezterm/config/lua/keyassignment/ActivateCommandPalette.html) like this:
![2024-01-03-134837@2x](https://github.com/wez/wezterm/assets/79978224/76102a2f-0caa-427d-b8fa-31b34ad4c9a8)

**Reply by eternity74** [reply]
Thanks!
How can i modify ssh config with enumerate_ssh_hosts ?
Do you mean do it from .wezterm.lua file ?
Or is it possible to do it on execution a command ?

**Comment by wez** [comment]
If the built-in ssh config parsing doesn't meet your needs, you may want to look at https://wezfurlong.org/wezterm/config/lua/keyassignment/InputSelector.html for a way to show a list of items to the user and have them select one.

If you want to allow the user to type in an ip address, then look at https://wezfurlong.org/wezterm/config/lua/keyassignment/PromptInputLine.html

**Reply by eternity74** [reply]
Thanks for the quick answer.

I tried to use InputSelector to get arbitrary hostname and connect ssh but I couldn't find any way to launch "wezterm ssh hostname" as new tab. What I found is that spawn tab by using ssh_domain with no multiplex. So I want to modify hostname part of existing ssh_domain for work around.

For input, I would prefer to pick one if possible from list but if it is not in the list, I would input directly on single ui. So I want to use like combo box.

Is there possible way to do it ?

**Reply by wez** [reply, marked as answer]
> couldn't find any way to launch "wezterm ssh hostname" as new tab.

There isn't currently a way to dynamically add an ssh domain for this, so your options are either to spawn it as a separate process with its own window, or spawn the `ssh` executable into a new tab/pane.  The latter is fine on unix systems, but is less desirable on windows systems where you may be trying to avoid conpty.

> For input, I would prefer to pick one if possible from list but if it is not in the list, I would input directly on single ui. So I want to use like combo box.

What I'd suggest is to use `InputSelector` to show the pre-defined list, add an option that will then trigger `PromptInputLine` if you want to enter an IP.  wezterm doesn't support doing both list and line input at once.

**Reply by eternity74** [reply]
I am working on windows. So I prefer to use ssh client of wezterm instead of windows ssh client. 
Regarding input I will try your suggestion.
Thanks for your help.
